### PR TITLE
Benchmark for creating Calendars.

### PR DESCRIPTION
**Pull request by MarchingCube**
## What does the pull request do?

Running ControlCatalog to benchmark controls tends to be quite annoying since it is hard to keep it consistent. This PR prepares basic infra for having low overhead control benchmarks and creates one for the `Calendar` control (which is one of the heavier ones).

Results from my machine:
|         Method |     Mean |    Error |   StdDev |    Gen 0 |    Gen 1 |   Gen 2 | Allocated |
|--------------- |---------:|---------:|---------:|---------:|---------:|--------:|----------:|
| CreateCalendar | 24.14 ms | 0.337 ms | 0.315 ms | 843.7500 | 406.2500 | 62.5000 |   4.84 MB |

We surely have quite some work to do since we need 5 MB and 24 ms to create a single `Calendar` :)

In this benchmark we spend around 25% in GC as well.

### CODE REVIEW

**Comment by Gillibald** [comment]
The amount of memory consumed is probably just because of non lazy loaded content but I could be wrong. Still this is a good benchmark.

**Comment by kekekeks** [comment]
Also see https://github.com/AvaloniaUI/Avalonia/pull/2847

**Comment by MarchingCube** [comment]
@Gillibald What do you mean by non lazy loaded content?

@kekekeks Yeah, I've seen that PR and probably would have used that if it was merged in.

**Comment by Gillibald** [comment]
If I have 12 months per year and only display one month at a time I don't have to prepare all 12 months in advance to display them later in time. Most likely I will just display one month and that's it.

I suspect that the Calendar control is holding a lot of data that isn't needed for the current visual representation. I suspect it is preparing everything in advance and that's why it takes so much time to load and consumes so much memory.

Haven't had a deeper look at the code so I could be wrong.

**Comment by ahopper** [comment]
How does the 24ms compare to non stubbed out version?

**Comment by Gillibald** [comment]
Can we monitor the number of SKPaint instances that are created when running this benchmark?

**Comment by MarchingCube** [comment]
@ahopper Not sure if I can initialize Skia renderer in a benchmark if you mean that. And if you mean comparison with our default test stubs - Moq is few orders of magnitude slower and consumes way more memory.

**Comment by MarchingCube** [comment]
@Gillibald Initial test was that we spend quite some time applying styles, but yeah I am not sure how much work can be skipped in the `Calendar` control. At least with this benchmark you can just run it with the profiler and get decent results for further investigation.

**Comment by ahopper** [comment]
@MarchingCube I was interested because 24ms seemed a lot faster than my perceived performance in control catalog.  I wondered if the difference gave more clues as to areas to work on and am also cautious in case the benchmark was not realistic. A set of benchmarks like this should be a great help 👍

File: tests/Avalonia.Benchmarks/Layout/CalendarBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+    [MemoryDiagnoser, InProcess, IterationCount(16)]
+    public class CalendarBenchmark : IDisposable
+    {
+        private readonly IDisposable _app;
+        private readonly TestRoot _root;
+        
+        public CalendarBenchmark()
+        {
+            _app = UnitTestApplication.Start(TestServices.StyledWindow.With(renderInterface: new NullRenderingPlatform(), threadingInterface: new NullThreadingPlatform()));
```
Comment (grokys) [inline tests/Avalonia.Benchmarks/Layout/CalendarBenchmark.cs]: Tiny nit: this line is too long for the GitHub source view meaning scrolling is needed.

**Comment by MarchingCube** [comment]
@ahopper ControlCatalog spawns 7 calendars. For me it usually takes 300ms to open that page which averages to 42 ms per calendar (although that page does more and having more controls slows styling down as well).

**Review by grokys — APPROVED** [review]
This is great, should really help us track performance improvements.

**Comment by ahopper** [comment]
@MarchingCube you have a faster computer than me :) that all makes sense.

**Comment by MarchingCube** [comment]
@ahopper Right, maybe I should post my hardware with benchmarks as well :)

This one was run on my Ryzen 2700X at home, I could check at work on my Threadripper 2950X as well to see if there is noticeable difference.

**Pull request merged**